### DISCUSSION TITLE
`asset/source` has no default, but trying to use the default. Is this expected behavior?

### DISCUSSION THREAD

**Discussion by ShortArrow** [webpack/webpack#16760] [Q&A] [answered]
I want to make Text Asset Load PoC for `asset/source`.

[github.com/ShortArrow/poc_shell_temp_in_webpack](https://github.com/ShortArrow/poc_shell_temp_in_webpack/tree/0ea2318929ff6d0c33630c429fb0d1e082f53cbd)

## Features

This is a minimal configuration to confirm that a text file can be imported as a literal string in TypeScript.
Using [Source assets](https://webpack.js.org/guides/asset-modules/#source-assets).

- Commands
  - `> PoC Shell Temp: Hello World`
  - `> PoC Shell Temp: Load text`
  - `> PoC Shell Temp: Load from mod`
  
## Main Issue

Commands of this extension has fail.
Because, these are get undefined by import syntax.
`asset/source` has not my expected behavior.
After verification. I was reach these idea.

### Point

This is a part of auto genarated `dist/extension.js`.
Using `testtext_txt_1.default`, but `default` has `undefined`.

```js
/***/ }),
/* 2 */
/***/ ((module) => {

module.exports = "Hello text data raw binding to literal\r\n";

/***/ }),
/* 3 */
/***/ ((__unused_webpack_module, exports, __webpack_require__) => {


Object.defineProperty(exports, "__esModule", ({ value: true }));
const testtext_txt_1 = __webpack_require__(2);
exports["default"] = testtext_txt_1.default
```

This "generated `dist/extension.js`" and "have not `default`" is expected behavior?
I think better as like as bellow.
Or, my `webpack.config.js` or `tsconfig.json` are mistaking?

### Actual

```js
module.exports = "Hello text data raw binding to literal\r\n";
```

### Expect

```js
module.exports.default = "Hello text data raw binding to literal\r\n";
// or module.exports = { default: "Hello text data raw binding to literal\r\n"};
```

**Comment by alexander-akait** [comment]
But there is not `module.exports.default` in commonjs (yes, you can use `default`, but what is the point?), sounds like you compile your code to commonjs using ts and them using `import ... from ...` like with normal ESM files, and there is a problem with `default`

**Reply by alexander-akait** [reply]
> Or, my webpack.config.js or tsconfig.json are mistaking?

So I recommend to transpile code to ESM to use it in ESM

**Reply by alexander-akait** [reply]
Please read https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#im-having-problems-with-esm-and-typescript, there are a lot of small and good infromation about ESM and tooling

**Reply by alexander-akait** [reply]
Feel free to feedback

**Comment by ShortArrow** [comment, marked as answer]
It was fixed by [src/types/textfile.d.ts](https://github.com/ShortArrow/poc_shell_temp_in_webpack/blob/135938536165723cc9e9e1c2ea40e2ed1c8c22df/src/types/textfile.d.ts) to expected behavior.

```diff
declare module "*.txt" {
  const content: string;
-  export default content;
+  export = content;
}
```

So I was noticed my type declare knowledge was missing.
